src/App/Fossa/EmbeddedBinary.hs

```diff
@@ -0,0 +1,125 @@
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module App.Fossa.EmbeddedBinary
+  ( extractEmbeddedBinary,
+    cleanupExtractedBinaries,
+    withEmbeddedBinary,
+    dumpEmbeddedBinary,
+    toExecutablePath,
+    BinaryPaths,
+    withWigginsBinary,
+    withSyftBinary,
+    allBins,
+    PackagedBinary (..)
+  )
+where
+
+import Control.Effect.Exception (bracket)
+import Control.Effect.Lift (Has, Lift, sendIO)
+import Control.Monad.IO.Class
+import Data.ByteString (ByteString, writeFile)
+import Data.FileEmbed.Extra
+import Path
+import Path.IO
+import Prelude hiding (writeFile)
+
+-- data BinaryPaths = BinaryPaths
+-- { binaryPathContainer :: Path Abs Dir
+-- , wigginsBinaryPath :: Path Abs File
+-- }
+
+data PackagedBinary
```

Comment:
This is a clever way of structuring this, and is probably good. I'm going to suggest another similar way to structure this, but don't feel obligated to change anything.

Rather than using `PackagedBinary` as a key for other functions like `extractedPath`, we can directly include that information in a record.

I would use this structure for PackagedBinary:
```hs
data PackagedBinary
  { packagedBinaryName :: Text
  , packagedBinaryBS :: ByteString
  }
```

and then we could replace, e.g.,
```hs
embeddedBinaryWiggins :: ByteString
```

with 
```hs
embeddedBinaryWiggins :: PackagedBinary
```

---

I understand this loses us the clever `allBins` implementation, and compile-time warnings if we forget to include something in there, so I'm kind of split.

Reply:
Hmm, I'm not sure I understand the advantage of doing it the way you proposed.  Can you explain a little more in detail **why** we might want to do it that way?